src/Database/Expression/QueryExpression.php

```diff
@@ -469,6 +469,22 @@ public function iterateParts(callable $callable)
     }
 
     /**
+     * Helps calling the `and()` and `or()` methods transparently.
+     *
+     * @param string $method The method name.
+     * @param array $args The argumemts to pass to the method.
+     * @return \Cake\Database\Expression\QueryExpression
+     * @throws \BadMethodCallException
+     */
+    public function __call($method, $args)
+    {
+        if (in_array($method, ['and', 'or'])) {
+            return call_user_func_array([$this, $method . '_'], $args);
+        }
+        throw new \BadMethodCallException;
```

Comment:
Including an exception message stating the invalid method name would be nice. Makes debugging easier.